### ISSUE TITLE
Back to Top button refreshes the feed

### ISSUE THREAD

**Issue by jordan1776** [liftoff-app/liftoff#349]
### Device info

- OS: Android
- OS version: 11
- Device: Xperia 1 II
- Liftoff version: 0.10.7

### Describe the bug

The "back to top" button in the feed view refreshes the page, exactly like the "refresh" button.

### Steps to reproduce

1. Swipe down the post view
2. Click on the 3 dot menu at the top
3. Tap the "back to top" button.

### Expected behavior

The button should simply go back to the top of the page without refreshing.

**Comment by Inovvia** [comment]
That is not a bug. That is how they currently implemented it.


```
if (value == 0) {
                      // unable to tie a controller to the infinite scroll
                      // table, for now just reload
                      // isc.scrollToTop();
                      isc.clear();
                    } else if (value == 1) {
                      isc.clear();
                    } else if (value == 2) {
                      goTo(context, (_) => const SettingsPage());
                    }
```

**Comment by jordan1776** [comment]
It is still a bug, just a usability one rather than runtime error

**Comment by mykdavies** [comment]
> It is still a bug, just a usability one rather than runtime error

Fair point 😄, looks like something that could be changed. Could you just help me understand what the use case is for "go to top but don't refresh"? It's not something that I get right now.

**Comment by jordan1776** [comment]
Same as the back to top on the comments view. So you can look at old posts that are still cached. Most apps have this.

**Comment by keforbes** [comment]
> Could you just help me understand what the use case is for "go to top but don't refresh"?

I would argue the label is misleading.  There's already a "Refresh" button next to "Back To Top", so why would "Back To Top" _also_ perform a refresh?  If I wanted to do a refresh, I would've clicked that button instead.  Also, the word "Back" wouldn't imply I'm getting "new" items.  "Back To Top" would make me think I'll go back to what the top looked like when I opened the app, while performing a refresh could give me an entirely different list of items.

Now, you could argue that "Refresh" is also implicitly doing a "Back To Top"... but the important thing there is I expect to find new items, and chronologically we put "new" at the top already.  So I don't _think_ I'm being contradictory here. :smile:

**Comment by sling-shot** [comment]
As of now there is another reason for this. The option to change the feed from say Active to Hot is at the top and scrolls away to become inaccessible. Only going back to the top can get it back into view.

**Comment by zachatrocity** [comment]
Because we're using an infinite scroll the posts are fetched. So when you scroll for awhile the posts above are refetched.

There's a couple things coming out of this:

1. Fix so you can access the hot/active filter without having to scroll to the top but rather as a scroll sliver so it shows like the 3-dot menu
2. If we really want to implement scroll to top then we need to cache the posts and update the infinite scroll widget to handle this scenaroo

**Comment by sling-shot** [comment]
My I suggest:

Put the two icons presently in the top 2nd row on the auto-hiding top row. Place the sorting button as icon between the bell icon and account instance button? (eg. All@lemmyworld). Place the view type icon next to 3-dot icon.

This might shrink the account instance button and render it unclear (because some of the text may not fit) so highlight the present selection in the pop-up list of accounts/instances that appear on touching that button.

We will have a single auto-hiding top bar which can serve the purpose.

NOTE: To solve the issue for those who do not want to refresh the feed, it will still need caching.

**Comment by keforbes** [comment]
> If we really want to implement scroll to top then we need to cache the posts and update the infinite scroll widget to handle this scenario

If we no longer have the "top" in memory and have to re-fetch it anyway, then there's really no functional difference between "Back To Top" and "Refresh".  If the only real reason to perform a "Back To Top" is to change the sorting type then I agree, we should just add a button (or menu) to change the sorting type which is always visible.  If we do that, we should just remove the "Back To Top" button entirely and only have "Refresh".

This may just be my use case, but I have no actual reason to return to the top of the feed and I don't think it'd be a good idea to cache every single post the user has seen _just in case_ someone wants to scroll to the top.  I say remove "Back To Top" and just make the sorting options always available.

**Comment by sling-shot** [comment]
In that case returning to the first post would only be needed if you now want to open it but a refresh may have lost it in case of volatile sorting like New.

**Comment by jordan1776** [comment]
>This may just be my use case, but I have no actual reason to return to the top of the feed and I don't think it'd be a good idea to cache every single post the user has seen just in case someone wants to scroll to the top. 

It might be a niche feature, but it is useful if you offline but still want to see something. I don't know how difficult to implement caching, it wouldn't need to be as advanced as Sync for Reddit's where it literally saves every position in the comments (which survives refresh and closing the app) and has a "restore position" snackbar for the feed.

I'm sure those buttons that sit above the first post would be made more accessible soon enough.
